src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -298,13 +299,26 @@ public OicSecurityRealm(
             throws IOException {
         // Needed in DataBoundSetter
         this.disableSslVerification = Util.fixNull(disableSslVerification, Boolean.FALSE);
+        if (FIPS140.useCompliantAlgorithms() && this.disableSslVerification) {
+            throw new IllegalArgumentException(Messages.OicSecurityRealm_DisableSslVerificationFipsMode());
+        }
         this.clientId = clientId;
         this.clientSecret = clientSecret;
         this.serverConfiguration = serverConfiguration;
     }
 
     @SuppressWarnings("deprecated")
     protected Object readResolve() throws ObjectStreamException {
+        // Fail if migrating to a FIPS non-compliant config
+        if (FIPS140.useCompliantAlgorithms()) {
+            if (isDisableSslVerification()) {
+                throw new IllegalArgumentException(Messages.OicSecurityRealm_DisableSslVerificationFipsMode());
+            }
+            if (isDisableTokenVerification()) {
+                throw new IllegalArgumentException(Messages.OicSecurityRealm_DisableTokenVerificationFipsMode());
+            }
+        }
```

Comment:
what's the correct exception for this method? Should be FormException as well? I'm tempted to say `IllegalStateException`
```suggestion
        if (FIPS140.useCompliantAlgorithms()) {
            if (isDisableSslVerification()) {
                throw new IllegalStateException(Messages.OicSecurityRealm_DisableSslVerificationFipsMode());
            }
            if (isDisableTokenVerification()) {
                throw new IllegalStateException(Messages.OicSecurityRealm_DisableTokenVerificationFipsMode());
            }
        }
```

Reply:
I guess both would be ok, just using it as it's the normally used exception (e.g.: https://github.com/jenkinsci/ldap-plugin/blob/master/src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java#L195-L202)

Reply:
no strong opinion